### ISSUE TITLE
vulkan overlay segfaults

### ISSUE THREAD

**Issue by sylware** [ValveSoftware/steam-for-linux#5568]
AMD GPU tahiti xt
AMD CPU FX9590
linux + mesa vulkan + drm, git no older than 1 week

In kernel log, I get:
[ 8820.417960] gameoverlayui[1864]: segfault at 204 ip 00000000f7fd4c55 sp 00000000ffd76430 error 4 in ld-2.27.so[f7fc0000+26000]
[ 8820.417969] Code: 00 00 8d 76 00 57 56 53 8b 74 24 10 e8 3b 69 00 00 81 c3 c4 23 01 00 83 ec 0c 8d bb 04 05 00 00 57 ff 93 34 08 00 00 83
c4 10 <f6> 86 04 02 00 00 08 75 18 8b 86 90 01 00 00 85 c0 74 20 83 ec 08
[ 8820.512993] gameoverlayui[1865]: segfault at 204 ip 00000000f7f1cc55 sp 00000000ff82e860 error 4 in ld-2.27.so[f7f08000+26000]
[ 8820.513000] Code: 00 00 8d 76 00 57 56 53 8b 74 24 10 e8 3b 69 00 00 81 c3 c4 23 01 00 83 ec 0c 8d bb 04 05 00 00 57 ff 93 34 08 00 00 83
c4 10 <f6> 86 04 02 00 00 08 75 18 8b 86 90 01 00 00 85 c0 74 20 83 ec 08
[ 8820.613862] gameoverlayui[1868]: segfault at 204 ip 00000000f7eecc55 sp 00000000ffbb3010 error 4 in ld-2.27.so[f7ed8000+26000]
[ 8820.613868] Code: 00 00 8d 76 00 57 56 53 8b 74 24 10 e8 3b 69 00 00 81 c3 c4 23 01 00 83 ec 0c 8d bb 04 05 00 00 57 ff 93 34 08 00 00 83
c4 10 <f6> 86 04 02 00 00 08 75 18 8b 86 90 01 00 00 85 c0 74 20 83 ec 08
[ 8820.710991] gameoverlayui[1869]: segfault at 204 ip 00000000f7f1cc55 sp 00000000ff966340 error 4 in ld-2.27.so[f7f08000+26000]
[ 8820.710998] Code: 00 00 8d 76 00 57 56 53 8b 74 24 10 e8 3b 69 00 00 81 c3 c4 23 01 00 83 ec 0c 8d bb 04 05 00 00 57 ff 93 34 08 00 00 83
c4 10 <f6> 86 04 02 00 00 08 75 18 8b 86 90 01 00 00 85 c0 74 20 83 ec 08
[ 8820.810690] gameoverlayui[1870]: segfault at 204 ip 00000000f7f04c55 sp 00000000ff8f4a80 error 4 in ld-2.27.so[f7ef0000+26000]
[ 8820.810696] Code: 00 00 8d 76 00 57 56 53 8b 74 24 10 e8 3b 69 00 00 81 c3 c4 23 01 00 83 ec 0c 8d bb 04 05 00 00 57 ff 93 34 08 00 00 83
c4 10 <f6> 86 04 02 00 00 08 75 18 8b 86 90 01 00 00 85 c0 74 20 83 ec 08
[ 8820.907743] gameoverlayui[1871]: segfault at 204 ip 00000000f7f7cc55 sp 00000000fff989c0 error 4 in ld-2.27.so[f7f68000+26000]
[ 8820.907751] Code: 00 00 8d 76 00 57 56 53 8b 74 24 10 e8 3b 69 00 00 81 c3 c4 23 01 00 83 ec 0c 8d bb 04 05 00 00 57 ff 93 34 08 00 00 83
c4 10 <f6> 86 04 02 00 00 08 75 18 8b 86 90 01 00 00 85 c0 74 20 83 ec 08

**Comment by kisak-valve** [comment]
Hello @sylware, please copy your system information from steam (```Steam``` -> ```Help``` -> ```System Information```) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report. If you find any CrashIDs or minidumps (commonly found in `/tmp/dumps`) related to this issue, please put them in an archive and drag and drop the archive into the comment box.

**Comment by sylware** [comment]
https://gist.github.com/0b7c65a902c0e61dc98278341c2179f5.git

**Comment by matthew-cline** [comment]
I'm also having a problem with vulkan, but it's the `vulkandriverquery` executable that's the problem.  Here's my [system info gist](https://gist.github.com/matthew-cline/172cce65cb4d92e3447f1c9485bebf95).  Trying to run that executable gives:

    Fatal : VkResult is "ERROR_INITIALIZATION_FAILED" in /home/pgriffais/src/Vulkan/base/vulkanexamplebase.cpp at line 823
    vulkandriverquery: /home/pgriffais/src/Vulkan/base/vulkanexamplebase.cpp:823: void VulkanExampleBase::initVulkan(): Assertion `res == VK_SUCCESS' failed.

When I try to run the command `vulkaninfo`, I get:

```
===========
VULKAN INFO
===========

Vulkan Instance Version: 1.1.77

ERROR: [Loader Message] Code 0 : /usr/lib/libvulkan_intel.so: wrong ELF class: ELFCLASS32
ERROR: [Loader Message] Code 0 : /usr/lib/libvulkan_radeon.so: wrong ELF class: ELFCLASS32
vulkaninfo/vulkaninfo.c:3214: failed with VK_ERROR_INITIALIZATION_FAILED
```

So for some reason the 64 bit vulkan code is trying to load the 32 bit vulkan drivers.

**Comment by sylware** [comment]
The opengl overlay is segfaulting the same way.
I don't have a javascript browser (I use lynx or links2), and will never use one. I cannot renew my dota+ subscription in dota or in the steam client...
(I switched to the beta client to see if that could help, it did not change anything)

**Comment by sylware** [comment]
I have a hunch: since 32bits is zillions times less tested than 64bits, I think gcc 8.1.0 is mis-compiling the 32 bits stack required for the remaining 32 bits parts of the steam client.
I'll recompile that 32bits stack with gcc 7.3.0 (I can easily because I use my custom gnu/linux distro)

**Comment by sylware** [comment]
nope, the overlay does segfault the same way with gcc-7.3.0.

**Comment by sylware** [comment]
I did narrow down the issue: a 32 bits gl/x11 libs incompatibilities between recent gl/x11 libs and the steam runtime libs.
If I want the client to start, I must use the recent mesa gl with the steam runtime x11 libs, but if I want the overlay to start, I must use the recent mesa gl with its recent x11 libs... if I start the client with the recent mesa gl with its recent x11 libs, it will crash...

Then I have to start the client, without any pinned 32 bits x11 libs, once started, I have to pinned the 32 bits x11 libs before starting a game in order to run the overlay without crashing.

I'll setup a full set of recent 32 bits x11 libs (not only the mesa gl required ones) to see if I can start the client with that.

**Comment by sylware** [comment]
I added most of x11 libs in recent version, 32 bits, now I have both the client which starts and the overlay running... but I have to remove config.vdf before starting steam and each time I have to log in with my password (if it keeps happening I'll open a new issue)

**Comment by sylware** [comment]
this issue is fixed on my system

**Comment by kisak-valve** [comment]
Closing per the last comment.
